tests/instruments/hp/test_hp33120A.py

```diff
@@ -0,0 +1,224 @@
+import pytest
+
+from pymeasure.test import expected_protocol
+from pymeasure.instruments.hp import HP33120A
+
+
+def test_init():
+    with expected_protocol(
+            HP33120A,
+            [(b'SOUR:VOLT:UNIT VPP', None)],
+    ):
+        pass  # Verify the expected communication.
+
+
+def test_amplitude_setter():
+    with expected_protocol(
+            HP33120A,
+            [(b'SOUR:VOLT:UNIT VPP', None),
+             (b'SOUR:VOLT 4', None)],
+    ) as inst:
+        inst.amplitude = 4
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'SOUR:VOLT?', b'+4.00000E+00')],
+     4.0),
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'SOUR:VOLT?', b'+4.00000E+00')],
+     4.0),
+))
+def test_amplitude_getter(comm_pairs, value):
+    with expected_protocol(
+            HP33120A,
+            comm_pairs,
+    ) as inst:
+        assert inst.amplitude == value
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'SOUR:VOLT:UNIT?', b'VPP')],
+     'Vpp'),
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'SOUR:VOLT:UNIT?', b'VPP')],
+     'Vpp'),
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'SOUR:VOLT:UNIT?', b'VPP')],
+     'Vpp'),
+))
+def test_amplitude_units_getter(comm_pairs, value):
+    with expected_protocol(
+            HP33120A,
+            comm_pairs,
+    ) as inst:
+        assert inst.amplitude_units == value
+
+
+def test_burst_count_setter():
+    with expected_protocol(
+            HP33120A,
+            [(b'SOUR:VOLT:UNIT VPP', None),
+             (b'BM:NCYC 500', None)],
+    ) as inst:
+        inst.burst_count = 500
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'BM:NCYC?', b'+1.00000E+00')],
+     1.0),
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'BM:NCYC?', b'+5.00000E+02')],
+     500.0),
+))
+def test_burst_count_getter(comm_pairs, value):
+    with expected_protocol(
+            HP33120A,
+            comm_pairs,
+    ) as inst:
+        assert inst.burst_count == value
+
+
+def test_burst_enabled_setter():
+    with expected_protocol(
+            HP33120A,
+            [(b'SOUR:VOLT:UNIT VPP', None),
+             (b'BM:STATE 1', None)],
+    ) as inst:
+        inst.burst_enabled = True
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'BM:STATE?', b'1')],
+     True),
+    ([(b'SOUR:VOLT:UNIT VPP', None),
+      (b'BM:STATE?', b'1')],
+     True),
```

Comment:
I guess, you want to test both cases: true and false.
Here you test true two times!

Reply:
thank you for catching this.
resolved in https://github.com/pymeasure/pymeasure/pull/1056/commits/ffc63b0b4b04c0de094fa1465dcd5d18aae1c0c2